src/pyhf/optimize/mixins.py

```diff
@@ -0,0 +1,140 @@
+        Create an optimizer.
+
+        Args:
+            maxiter (`int`): maximum number of iterations. Default is 100000.
+            verbose (`bool`): print verbose output during minimization. Default is off.
+        """
+        self.maxiter = kwargs.pop('maxiter', 100000)
+        self.verbose = kwargs.pop('verbose', False)
+        self._minimizer = None
+
+        if kwargs:
+            raise KeyError(
+                f"""Unexpected keyword argument(s): '{"', '".join(kwargs.keys())}'"""
+            )
+
+    def _internal_minimize(
+        self,
+        func,
+        init,
+        method='SLSQP',
+        jac=None,
+        bounds=None,
+        fixed_vals=None,
+        options={},
+    ):
+        result = self._minimize(
+            func,
+            init,
+            method=method,
+            bounds=bounds,
+            fixed_vals=fixed_vals,
+            options=options,
+            jac=jac,
+        )
+
+        try:
+            assert result.success
+        except AssertionError:
+            log.error(result)
+            raise
+        return result
+
+    def minimize(
+        self,
+        objective,
+        data,
+        pdf,
+        init_pars,
+        par_bounds,
+        fixed_vals=None,
+        return_fitted_val=False,
+        do_grad=False,
+        do_stitch=False,
+        method='SLSQP',
+        **kwargs,
+    ):
+        """
+        Find Function Parameters that minimize the Objective.
+
+        Args:
+            objective: objective function
+            data: observed data
+            pdf: model
+            init_pars: initial parameters
+            par_bounds: parameter boundaries
+            fixed_vals: fixed parameter values
+            return_fitted_val: return bestfit value of the objective
+            do_grad (`bool`): enable autodifferentiation mode. Default is off.
+            do_stitch (`bool`): enable splicing/stitching fixed parameter.
+            method: minimization routine
+            kwargs: other options to pass through to underlying minimizer
+
+        Returns:
+            bestfit parameters
+
+        """
+        tensorlib, _ = get_backend()
+        tv, fixed_values_tensor, func_and_grad, init, bounds = shim(
+            objective,
+            data,
+            pdf,
+            init_pars,
+            par_bounds,
+            fixed_vals,
+            do_grad=do_grad,
+            do_stitch=do_stitch,
+        )
+        if do_grad:
+            func = lambda pars: func_and_grad(pars)[0]
+            jac = lambda pars: func_and_grad(pars)[1]
+        else:
+            func = func_and_grad
+            jac = None
+
+        self._setup_minimizer(func, data, pdf, init_pars, par_bounds, fixed_vals)
+
+        minimizer_kwargs = dict(method=method, bounds=bounds, options=kwargs, jac=jac)
+        if not do_stitch:
+            minimizer_kwargs.update(dict(fixed_vals=fixed_vals))
+        result = self._internal_minimize(func, init, **minimizer_kwargs)
+
```

Comment:
we need to somehow bridge the gap that minuit is stateful and scipy is stateless

```
self. _setup_minimizer(func, data, pdf, init_pars, par_bounds, fixed_vals) #prep self._minimizer obj
self._minimize() #use self.minimizer
```

could wrap scipy.optimize call into a stateless thin

```
self.minimizer = lambda : scipy.optimize.minimize(...)
```

which in `_internal_minimize()` might just be used via `self.minimizer()`